avocado/utils/pci.py

```
@@ -376,6 +376,96 @@ def get_vendor_id(full_pci_address):
+    return any("MSI-X:" in line for line in out.stdout_text.splitlines())
+
+
+def device_supports_irqs(full_pci_address, count):
+    """
+    Check if the device supports at least the specified number of interrupts.
+
+    :param full_pci_address: Full PCI device address including domain (e.g., 0000:03:00.0)
+    :param count: Number of IRQs the device should support
+    :return: True if supported, False otherwise
+    """
+    out = process.run(
```

Ruff (0.14.0)

Function call with `shell=True` parameter identified, security issue

(S604)